version/validate_compatibility.go

```diff
@@ -0,0 +1,99 @@
+package version
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/astronomer/astro-cli/deployment"
+	"github.com/pkg/errors"
+
+	"github.com/Masterminds/semver"
+	"github.com/astronomer/astro-cli/houston"
+	"github.com/astronomer/astro-cli/messages"
+)
+
+// ValidateCompatibility print message if astro-cli version is not compatible with platform version
+func ValidateCompatibility(client *houston.Client, out io.Writer, cliVer string) error {
+	serverCfg, err := deployment.AppConfig(client)
+	if err != nil {
+		return err
+	}
+	// Skip check if AppConfig is nil or is cv is empty
+	if serverCfg != nil && cliVer != "" {
+		return compareVersions(serverCfg.Version, cliVer, out)
+	}
+	return nil
+}
+
+func compareVersions(serverVer string, cliVer string, out io.Writer) error {
+	if isBehindMajor(serverVer, cliVer) {
+		return errors.Errorf(messages.ERROR_NEW_MAJOR_VERSION, cliVer, serverVer)
```

Comment:
@adam2k  pay attention here, i prefer to not crash just notify user.
@schnie @ashb thoughts?

Reply:
@andriisoldatenko thanks!  Yeah, I wanted to get feedback to see what everybody thinks is the best approach.  I can revert this change if most people don't like the error here.

Reply:
The UX for an error now is that there will be a graphQL error if old CLI that is incompatible with server or vice versa. This usually is resolved by a support ticket. Perhaps we replace graphQL errors with an upgrade error to be more informative.

In terms of enforce by default, my main concern is CI / automation users who are using an old minor version CLI that does still work with a new version. I prefer enforce by default when we have a compatibility matrix, and it's enforced with no --skip option for situations we know will break. This way we avoid breaking users where it happens to work between minor versions and we always stop behavior we know will break.